Hcal/python/TrainVisiblesBDT.py

```diff
@@ -0,0 +1,321 @@
+        sigWgt = np.zeros(len(sig)) + 1
+        bkgWgt = np.zeros(len(bkg)) + 1.*float(len(sig))/float(len(bkg))
+        return np.append(sigWgt, bkgWgt)
+
+if __name__ == '__main__' :
+
+    parser = OptionParser()
+
+    parser.add_option('--seed', dest='seed', type='int', default=4, help='Numpy random seed.')
+    parser.add_option('--train_frac', dest='train_frac', default=0.9999, help='Fraction of events to use for training')
+    parser.add_option('--max_evt', dest='max_evt', type='int', default=3000000, help='Max Events to load')
+    parser.add_option('--out_dir', dest='out_dir', default='~/', help='Output directory')
+    parser.add_option('--out_name', dest='out_name', default='bdt', help='Output Pickle Name')
+    parser.add_option('--eta', dest='eta', type='float', default=0.023, help='Learning Rate')
+    parser.add_option('--tree_number', dest='tree_number', type='int', default=1000, help='Tree Number')
+    parser.add_option('--depth', dest='depth', type='int', default=6, help='Max Tree Depth')
+    parser.add_option('--bkg_dir', dest='bkg_dir', default='~/', help='name of background file directory')
+    parser.add_option('--sig_dir', dest='sig_dir', default='~/', help='name of signal file directory')
+
+    (options, args) = parser.parse_args()
+
+    np.random.seed(options.seed)
+
+    adds = 0
+    Check = True
+    while Check :
+        if not os.path.exists(options.out_name+'_'+str(adds)) :
+            try :
+                os.makedirs(options.out_name+'_'+str(adds))
+                Check = False
+            except :
+                Check = True
+            else :
+                adds += 1
+
+    print("Random seed is", options.seed)
+    print("Tree number:", options.tree_number)
+    print("Max tree depth:", options.depth)
+    print("Eta:", options.eta)
+    print("Max training events:", options.max_evt)
+    print("Training fraction:", options.train_frac)
+
+    print("Loading signal files from", options.sig_dir)
+    sigContainer = sampleContainer(options.sig_dir, options.max_evt, options.train_frac, False)
+    sigContainer.constructTrainAndTest()
+
+    print("Loading background files from", options.bkg_dir)
+    bkgContainer = sampleContainer(options.bkg_dir, options.max_evt, options.train_frac, True)
+    bkgContainer.constructTrainAndTest()
+
+    mergedContainer = mergedContainer(sigContainer, bkgContainer)
+
+    params = {"objective": "binary:logistic",
+              "eta": options.eta,
+              "max_depth": options.depth,
+              "min_child_weight": 20,
+              "silent": 0,
+              "subsample": 0.9,
+              "colsample_bytree": 0.85,
+              "eval_metric": 'auc',
+              "seed": 1,
+              "nthread": 1,
+              "verbosity": 1,
+              "early_stopping_rounds": 10}
+
+    evallist = [(eventContainer.dtest, 'eval'), (eventContainer.dtrain, 'train')]
+    gbm = xbg.train(params, eventContainer.dtrain, options.tree_number, evallist)
```

Comment:
```suggestion
    gbm = xgb.train(params, eventContainer.dtrain, options.tree_number, evallist)
```
Did the original really work?

Reply:
I have been neglecting this file. I will overhaul it with our group's up-to-date training method, which doesn't rely on looping over events in Python! Does a script for training the visibles BDT belong in the Hcal package or is there a better place for it? It could go outside of `ldmx-sw` and instead go in `ldmx-publications` as an analysis script, like [here](https://github.com/LDMX-Software/ldmx-publications/tree/2025visibles/2025-visibles/analysis-code) (more will change in that repo to reflect how this PR is integrated into `ldmx-sw`).

Reply:
I think it makes more sense for it to be outside ldmx-sw since it is not related to the C++ processing (but we also have other stuff that's not related to the C++ processing floating around).